webpack/private/webpack.config.js.tmpl

```diff
@@ -1,22 +1,31 @@
-module.exports = {
-  infrastructureLogging: {
+module.exports = function () {
+  const v4 = new Error().stack.includes('webpack-cli@4.')
+
+  const infrastructureLogging = v4 ? {level: 'error'} : {
     colors: false,
     console: new console.Console(process.stderr, process.stderr),
     level: 'error',
-  },
+  }
+
   // We want webpack to spit out deterministic file hashes and module ids
   // this will improve remote caching
   // https://webpack.js.org/guides/caching/#module-identifiers
-  optimization: {
-    moduleIds: 'deterministic',
-    chunkIds: 'deterministic',
-  },
-  output: {
+  const optimization = {
+    moduleIds: v4 ? 'natural' : 'deterministic',
+    chunkIds: v4 ? 'natural' : 'deterministic',
+  }
+
+  // Webpack4: passing any 'output' overrides the standard output config in entries
+  const output = v4 ? {
+    path: __dirname,
```

Comment:
Without this it outputs to `dist/` by default, the webpack5 `uniqueName` must override the default `dist/`? 🤷

Reply:
what about `--output` flag we specify? doesn't that override this?

Reply:
You would think so, but no it doesn't seem like it :(